File: src/bmsx_cpp/render/2d/vdp_blit_pipeline.cpp
#include "render/2d/vdp_blit_pipeline.h"

#include "render/backend/backend.h"
#include "render/backend/pass/framebuffer_execution.h"
#include "render/backend/pass/library.h"
#include <any>

namespace bmsx {

void registerVdpFrameBufferExecutionPass(RenderPassLibrary& registry) {
	RenderPassDef desc;
	configureVdpFrameBufferExecutionPass(desc);
	desc.exec = [](GPUBackend* backend, void*, std::any& state) {
		auto& executionState = std::any_cast<VdpFrameBufferExecutionPassState&>(state);
		backend->executeVdp2DBlit(executionState);
	};
	registry.registerPass(desc);
}

} // namespace bmsx

File: src/bmsx/render/2d/vdp_blit_pipeline.ts
import type { VdpFrameBufferExecutionPassState } from '../backend/backend';
import type { RenderPassLibrary } from '../backend/pass/library';

export function registerVdpFrameBufferExecutionPass(registry: RenderPassLibrary): void {
	registry.register<VdpFrameBufferExecutionPassState>({
		id: 'vdp_framebuffer_execution',
		name: 'VDP2DBlitExecution',
		stateOnly: true,
		graph: { skip: true },
		bootstrap: (backend) => {
			backend.bootstrapVdp2DBlit();
		},
		exec: (backend, _fbo, state) => {
			backend.executeVdp2DBlit(state);
		},
	});
}
